packages/subgraph/src/utils/tokens/erc1155.ts

```diff
@@ -0,0 +1,349 @@
+import {Address, BigInt, Bytes, ethereum, log} from '@graphprotocol/graph-ts';
+import {
+  ERC1155Balance,
+  ERC1155Contract,
+  ERC1155TokenIdBalance,
+  ERC1155Transfer
+} from '../../../generated/schema';
+import {ERC1155} from '../../../generated/templates/DaoTemplateV1_0_0/ERC1155';
+import {supportsInterface} from '../erc165';
+import {
+  DECODE_OFFSET,
+  ERC1155_MAGIC_NUMBER,
+  ERC1155_safeBatchTransferFrom,
+  ERC1155_safeTransferFrom,
+  ERC165_MAGIC_NUMBER,
+  TransferType,
+  getBalanceId,
+  getERC1155TransferId,
+  getTokenIdBalanceId
+} from './common';
+
+export function supportsERC1155(token: Address): bool {
+  // Double check that it's ERC1155 by calling supportsInterface checks.
+  let erc1155 = ERC1155.bind(token);
+  let introspection_ERC165 = supportsInterface(erc1155, ERC165_MAGIC_NUMBER); // ERC165
```

Comment:
Minor thing:
The correct term here is "interfaceId", not "magic number". 
You obtain these numbers in the contracts via `type(IERC165).interfaceId` or `type(IERC1155).interfaceId`

Reply:
👍  should I change the names to XXX_INTERFACE_ID then?